shared/metrics/metrics.go

```diff
@@ -55,117 +65,211 @@ func FromContext(ctx context.Context) Metrics {
 var _ Metrics = (*PromMetrics)(nil)
 
 type PromMetrics struct {
-	gasBalance                    metrics.Gauge
-	totalSolverLoops              metrics.Counter
```

Comment:
same with this metric, we would need to add a db method to get individual orders i believe.. unless i am misunderstanding what a solverLoop is